### ISSUE TITLE
yum clean all should not be required when not in last build stage

### ISSUE THREAD

**Issue by akostadinov** [hadolint/hadolint#929]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

ignore DL3032 when not in the latest stage of build

### Actual behavior

reports DL3032 in any stage of the build

### Steps to reproduce the behavior

```
$ cat Dockerfile
FROM registry.access.redhat.com/ubi7/ubi AS builder
RUN yum install -y gcc-c++ mysql-devel make

FROM registry.access.redhat.com/ubi7-minimal
COPY --from=builder /home/whatever /somewhere/else

$ hadolint Dockerfile
Dockerfile:2 DL3032 warning: `yum clean all` missing after yum command.
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
hadolint-2.8.0-8.fc36.x86_64
```
